src/main/java/net/dries007/tfc/objects/te/TEPowderKeg.java

```diff
@@ -0,0 +1,257 @@
+        markForSync();
+    }
+
+    public void onUnseal()
+    {
+        // Update sealed tick info and sync to client
+        sealedTick = sealedCalendarTick = 0;
+        sealed = false;
+        markForSync();
+    }
+
+    public boolean isSealed()
+    {
+        return sealed;
+    }
+
+    @Override
+    public void readFromNBT(NBTTagCompound nbt)
+    {
+        super.readFromNBT(nbt);
+        sealedTick = nbt.getLong("sealedTick");
+        sealedCalendarTick = nbt.getLong("sealedCalendarTick");
+        sealed = sealedTick > 0;
+    }
+
+    @Override
+    @Nonnull
+    public NBTTagCompound writeToNBT(NBTTagCompound nbt)
+    {
+        nbt.setLong("sealedTick", sealedTick);
+        nbt.setLong("sealedCalendarTick", sealedCalendarTick);
+        return super.writeToNBT(nbt);
+    }
+
+    @Override
+    public boolean hasCapability(Capability<?> capability, @Nullable EnumFacing facing)
+    {
+        return capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> T getCapability(Capability<T> capability, @Nullable EnumFacing facing)
+    {
+        if (capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY)
+        {
+            return (T) new ItemHandlerSidedWrapper(this, inventory, facing);
+        }
+        return super.getCapability(capability, facing);
+    }
+
+    @Override
+    public void onBreakBlock(World world, BlockPos pos, IBlockState state)
+    {
+        if (!state.getValue(SEALED))
+        {
+            // Not sealed, so empty contents normally
+            super.onBreakBlock(world, pos, state);
+        }
+        else
+        {
+            // Need to create the full barrel and drop it now
+            ItemStack stack = getItemStack(state);
+            InventoryHelper.spawnItemStack(world, pos.getX(), pos.getY(), pos.getZ(), stack);
+        }
+    }
+
+    @Override
+    public boolean isItemValid(int slot, ItemStack stack)
+    {
+        return stack.getItem() == Items.GUNPOWDER;
+    }
+
+    public void setIgniter(@Nullable EntityLivingBase igniterIn)
+    {
+        igniter = igniterIn;
+    }
+
+    /**
+     * Called to get the NBTTagCompound that is put on Barrel Items.
+     * This happens when a sealed Barrel was broken.
+     *
+     * @return An NBTTagCompound containing inventory and tank data.
+     */
+    private NBTTagCompound getItemTag()
+    {
+        NBTTagCompound nbt = new NBTTagCompound();
+        nbt.setTag("inventory", inventory.serializeNBT());
+        nbt.setLong("sealedTick", sealedTick);
+        nbt.setLong("sealedCalendarTick", sealedCalendarTick);
+        return nbt;
+    }
+
+    public int getStrength()
+    {
+        int count = 0;
+        for (int i = 0; i < inventory.getSlots(); i++)
+        {
+            count += inventory.getStackInSlot(i).getCount();
+        }
+        return count / 12;
+    }
+
+    public int getFuse() {
+        return fuse;
+    }
```

Comment:
formatting

Reply:
Gah! I ran the code style checker before I added this! :/